compiler/passes/checkParsed.cpp

```diff
@@ -479,6 +480,39 @@ static void nestedName(ModuleSymbol* mod) {
   }
 }
 
+static void includedStrictNames(ModuleSymbol* mod) {
+  if (mod->defPoint == NULL) {
+    return;
+  }
+
+  if (mod->hasFlag(FLAG_INCLUDED_MODULE)) {
+    ModuleSymbol* parent = mod->defPoint->getModule();
+
+    // module name should match file name
+    const char* fname = filenameToModulename(parent->astloc.filename);
+    if (fname != parent->name) {
+      USR_FATAL("Cannot include module from module with different file name");
+    }
+
+    // parent module must be top-level in its file.
+    // in is not necessarily a top-level module, though.
+    ModuleSymbol* lastParentSameFile = parent;
+    for (ModuleSymbol* cur = parent;
+         cur != NULL && cur->defPoint != NULL;
+         cur = cur->defPoint->getModule()) {
+      if (parent->astloc.filename == cur->astloc.filename) {
+        lastParentSameFile = cur;
+      } else {
+        break;
+      }
+    }
+
+    if (lastParentSameFile != parent) {
+      USR_FATAL(parent, "Cannot include module from a inline nested module");
```

Comment:
"an inline"? (and maybe even "an in-line" since I think "in-line" is serving as an adjective rather than a verb here?)

Reply:
fixed